test_integration/erc20_gateway/03_withdraw_and_confirm_withdraw/03_withdraw_and_confirm_withdraw.ts

```diff
@@ -0,0 +1,194 @@
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import BN from 'bn.js';
+
+import shared from "../shared";
+import Utils from '../Utils';
+import Assert from "../Assert";
+import Interacts from "../../../interacts/Interacts";
+
+describe('withdraw and confirm withdraw', async (): Promise<void> => {
+
+  let erc20Cogateway = shared.contracts.ERC20Cogateway;
+  let erc20Gateway = shared.contracts.ERC20Gateway;
+  let blockNumber: BN;
+  let utilityToken: string;
+  let withdrawMessageHash: string;
+  let withdrawParams: { withdrawalAmount: any; beneficiary: any; feeGasPrice: any; feeGasLimit: any; utilityToken: any; };
+
+  it('should withdraw successfully', async (): Promise<void> => {
+    utilityToken = await erc20Cogateway.instance.methods.utilityTokens(
+      shared.contracts.ValueToken.address,
+    ).call();
+    const utilityTokenInstance = Interacts.getUtilityToken(shared.web3, utilityToken);
+
+    const rawTx = utilityTokenInstance.methods.approve(
+      erc20Cogateway.address,
+      '100',
+    );
+
+    const withdrawerAddress = shared.beneficiaryAtAuxiliary;
+
+    await Utils.sendTransaction(
+      rawTx,
+      {
+        from: withdrawerAddress,
+      }
+    );
+
+    shared.beneficiaryAtOrigin = shared.accounts[11];
+    withdrawParams = {
+      withdrawalAmount: new BN(40),
+      beneficiary: shared.beneficiaryAtOrigin,
+      feeGasPrice: new BN(2),
+      feeGasLimit: new BN(10),
+      utilityToken: utilityToken,
+    };
+
+    const withdrawerBalanceBeforeWithdraw = await utilityTokenInstance.methods.balanceOf(
+      withdrawerAddress,
+    ).call();
+
+    const withdrawRawTx = erc20Cogateway.instance.methods.withdraw(
+      withdrawParams.withdrawalAmount.toString(10),
+      withdrawParams.beneficiary,
+      withdrawParams.feeGasPrice.toString(10),
+      withdrawParams.feeGasLimit.toString(10),
+      withdrawParams.utilityToken,
+    );
+
+    const tx = await Utils.sendTransaction(
+      withdrawRawTx,
+      {
+        from: withdrawerAddress,
+      }
+    );
+
+    withdrawMessageHash = tx.events.WithdrawIntentDeclared.returnValues['messageHash'];
+
+    const withdrawerBalanceAfterWithdraw = await utilityTokenInstance.methods.balanceOf(
+      withdrawerAddress,
+    ).call();
+
+    await Assert.assertWithdrawIntentDeclared(
+      tx.events.WithdrawIntentDeclared,
+      withdrawerAddress,
+      withdrawParams,
+    );
+
+    Assert.assertWithdraw(
+      new BN(withdrawerBalanceBeforeWithdraw),
+      new BN(withdrawerBalanceAfterWithdraw),
+      withdrawParams.withdrawalAmount,
+    );
+  });
+
+  it('should anchor successfully', async (): Promise<void> => {
```

Comment:
should anchor auxiliary stateroot on origin?

Reply:
Done.